app/models/entities/item.rb

```diff
@@ -1,59 +1,132 @@
-  #             },
-  #             connec_entity_names[1]: {
-  #               external_entities_names[0]: [unmapped_connec_entitiy3],
-  #               external_entities_names[1]: [unmapped_connec_entitiy4]
-  #             }
-  #          }
-  def connec_model_to_external_model!(connec_hash_of_entities)
-    items = connec_hash_of_entities['item']
-    connec_hash_of_entities['item'] = { 'Product2' => [], 'PricebookEntry' => [] }
-
-    items.each do |item|
-      connec_hash_of_entities['item']['Product2'] << item
-      connec_hash_of_entities['item']['PricebookEntry'] << item
+class Entities::Item < Maestrano::Connector::Rails::Entity
+
+  def connec_entity_name
+    'Item'
+  end
+
+  def external_entity_name
+    'Product'
+  end
+
+  def mapper_class
+    ItemMapper
+  end
+
+  def object_name_from_connec_entity_hash(entity)
+    entity['name']
+  end
+
+  def object_name_from_external_entity_hash(entity)
+    entity['title']
+  end
+
+
+  def consolidate_and_map_data(connec_entities, external_entities, organization, opts={})
+    items_with_variant = group_items_variants(connec_entities)
+    super(items_with_variant, external_entities, organization, opts)
+  end
+
+  def push_entities_to_connec(connec_client, mapped_external_entities_with_idmaps, organization)
+    # 1/ push the orders
+    self.push_entities_to_connec_to(connec_client, mapped_external_entities_with_idmaps, self.connec_entity_name, organization)
+    variants = []
+    mapped_external_entities_with_idmaps.each do |product|
+      parent_connect_id = product[:idmap].connec_id
+      product[:entity][:variants].each do |variant|
+        variant[:parent_item_id] = parent_connect_id
+        idmap = Maestrano::Connector::Rails::IdMap.find_by(external_id: variant[:external_id], connec_entity: connec_entity_name.downcase, external_entity: 'variant', organization_id: organization.id)
+        variants.push({entity: variant, idmap: idmap || create_id_map(variant, organization)})
+      end
     end
+    # 2/ push the variants
+    self.push_entities_to_connec_to(connec_client, variants, self.connec_entity_name, organization)
+  end
+
+  def push_entities_to_external(external_client, mapped_connec_entities_with_idmaps, organization)
+    mapped_connec_entities_with_idmaps.each do |mapped_connec_entity_with_idmap|
+      product = mapped_connec_entity_with_idmap[:entity]
+      product_id_map = mapped_connec_entity_with_idmap[:idmap]
+      product[:variants].each do |variant|
+        idmap = Maestrano::Connector::Rails::IdMap.find_by(connec_id: variant[:connec_id], connec_entity: connec_entity_name.downcase, external_entity: 'variant', organization_id: organization.id)
+        variant[:id] = idmap.external_id if idmap
+        variant[:product_id] = product_id_map.external_id
+      end
+    end
+    push_entities_to_external_to(external_client, mapped_connec_entities_with_idmaps, self.external_entity_name, organization)
```

Comment:
`super` instead of `push_entities_to_external_to(external_client, mapped_connec_entities_with_idmaps, self.external_entity_name, organization)`?